crates/oxc_linter/src/rules/oxc/only_used_in_recursion.rs

```diff
@@ -367,6 +636,80 @@ fn test() {
         };
         validator()
         "#,
+        // no params, no recursion
+        "
+           function Test() {
+                return <Test1 />;
+            }
+        ",
+        "
+            function Listitem({ depth }) {
+                console.log(depth);
+                return <Listitem depth={depth}/>;
+            }
+        ",
```

Comment:
can we add a case?
```ts
            function Listitem({ depth }) {
                 console.log(depth);
                 depth = depth + 1;
                 return <Listitem depth={depth}/>;
             }

Reply:
oh wait should this be a fail or pass case?

Reply:
Thank you for the suggestion!
I’ve added the requested pass for this case: 32f319e8693ba9ce943c057216966fc19bc74dc6

Reply:
The function below creates infinite recursion, which this rule should detect. 
However, the current implementation does not issue a warning for this case. 
I just tested this.

```
            function Listitem({ depth }) {
                 depth = depth + 1;
                 return <Listitem depth={depth}/>;
             }
```

Reply:
Do you have any ideas on how we could warn for this case?
The current implementation checks that:

The reference is inside the callee.
The argument is a single-node expression.
This case doesn’t pass the first condition, so we’d likely need to rethink the detection logic to catch it.

Reply:
lets tackle this in another PR and keep the scope of this one on the smaller side?

it'd probably be easier after this is solved first >> https://github.com/oxc-project/oxc/pull/7120#discussion_r1827918922

Reply:
nah this is fine for now